exif-images/functions/index.js

```diff
@@ -44,10 +49,13 @@ exports.metadata = functions.storage.object().onChange(event => {
     return;
   }
 
-  // Download file from bucket.
   const bucket = gcs.bucket(object.bucket);
-  return bucket.file(filePath).download({
-    destination: tempLocalFile
+  //create the temp directory where storage file will be downloaded
+  return mkdirp(tempLocalDir).then(()=>{
```

Comment:
first create the temp directory, and download the file after successful creation of the directory